build-scripts/build_tests.py

```diff
@@ -0,0 +1,230 @@
+                template_parameters = template.preprocess(
+                    rendered_rule_obj["template"]["vars"], "test")
+                env_yaml = env_yaml.copy()
+                jinja_dict = ssg.utils.merge_dicts(env_yaml, template_parameters)
+                file_contents = ssg.jinja.process_file_with_macros(str(test.absolute()),
+                                                                   jinja_dict)
+                scenario = tests.ssg_test_suite.rule.Scenario(test.name, file_contents)
+                if scenario.matches_platform(benchmark_cpes):
+                    rule_output_path.mkdir(parents=True, exist_ok=True)
+                    test_output_path = rule_output_path / test.name
+                    _write_path(file_contents, test_output_path)
+                    logger.debug("Wrote scenario %s for rule %s", test.name, rule_id)
+                else:
+                    logger.debug("Skipping scenario %s for rule %s has it not applicable",
+                                 test.name, rule_id)
+
+
+def _get_benchmark_cpes(env_yaml) -> Set[str]:
+    benchmark_cpes = set()
+    for cpe in env_yaml["cpes"]:
+        for _, data in cpe.items():
+            benchmark_cpes.add(data["name"])
+    return benchmark_cpes
+
+
+def _get_rules_in_profile(built_profiles_root) -> Generator[str, None, None]:
+    for profile_file in built_profiles_root.iterdir():  # type: pathlib.Path
+        if not profile_file.name.endswith(".profile"):
+            continue
+        profile_data = ssg.yaml.open_raw(str(profile_file.absolute()))
+        for selection in profile_data["selections"]:
+            if "=" not in selection:
+                yield selection
+
+
+def main() -> int:
+    args = _create_arg_parser().parse_args()
+    logging.basicConfig(level=logging.getLevelName(args.log_level))
+    env_yaml = ssg.environment.open_environment(args.build_config_yaml, args.product_yaml)
+    root_path = pathlib.Path(args.root).resolve().absolute()
+    output_path = pathlib.Path(args.output).resolve().absolute()
+    resolved_rules_dir = pathlib.Path(args.resolved_rules_dir)
+    if not resolved_rules_dir.exists() or not resolved_rules_dir.is_dir():
+        logging.error("Unable to find product at %s", str(resolved_rules_dir))
+        logging.error("Is the product built?")
+        return 1
+
+    output_path.mkdir(parents=True, exist_ok=True)
+    _copy_and_process_shared(env_yaml, output_path, root_path)
+    benchmark_cpes = _get_benchmark_cpes(env_yaml)
+
+    built_profiles_root = resolved_rules_dir.parent / "profiles"
+    rules_in_profiles = list(_get_rules_in_profile(built_profiles_root))
+
+    templates_root = root_path / "shared" / "templates"
+    all_resolved_rules = list(resolved_rules_dir.iterdir())
+    processes = list()
+    for chunk in range(args.jobs):
+        process_args = (benchmark_cpes, env_yaml, output_path,
+                        all_resolved_rules[chunk::args.jobs], templates_root, rules_in_profiles,)
```

Comment:
I would split the rules_in_profiles into chunks instead of all_resolved_rules because the all_resolved_rules contain rules that aren't going to be processed because they are not present in the built content so the subprocesses won't be load balanced. If you split rules_in_profiles to chunks each subprocess would get the same amout of rules to be processed, which would be better load balancing.